lib/routes/devices-and-sessions.js

```diff
@@ -237,6 +242,93 @@ module.exports = (log, db, config, customs, push, devices) => {
         }
       }
     },
+    {
+      method: 'GET',
+      path: '/account/device/messages',
+      config: {
+        validate: {
+          query: {
+            index: isA.string().alphanum().optional(),
+            limit: isA.number().optional().min(0).max(100),
+          }
+        },
+        auth: {
+          strategy: 'sessionToken'
+        },
+        response: {
+          schema: PUSHBOX_RETRIEVE_SCHEMA
+        }
+      },
+      handler (request, reply) {
+        log.begin('Account.deviceMessages', request)
+
+        const sessionToken = request.auth.credentials
+        const uid = sessionToken.uid
+        const deviceId = sessionToken.deviceId
+        const ip = request.app.clientAddress
+        const query = request.query || {}
+        const {index} = query
+        const limit = query.limit !== undefined ? Math.min(100, query.limit) : 100
+
+        return customs.checkAuthenticated('deviceMessages', ip, uid)
```

Comment:
I wonder if we need to rate-limit the "fetch" side of this, esp. if we expect clients to be polling it, the load on querying customs-server might outweigh any benefit of rate-limiting.  We should still rate-limit the sends, just not sure it's valuable for the receives.